tool/tsh/common/app.go

```diff
@@ -49,12 +50,17 @@ func onAppLogin(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
-	appInfo, err := getAppInfo(cf, tc, nil /*matchRouteToApp*/)
+	appInfo, clusterClient, err := getAppInfo(cf, tc, nil /*matchRouteToApp*/)
```

Comment:
I've placed it because we can reuse the `RetryWithRelogin` (which is always required on `getAppInfo` to fetch the profiles). Also, this is the first function executed by the app commands. It is convenient instead of creating the cluster client outside for each app command.

Reply:
After some testing, I noticed this update caused some cases where login is required (For example, if the device trust configurations change). For those cases, the `ConnectToCluster` operation will succeed (given it just creates the gRPC clients and doesn't immediately connect), but the `GetEnrichedResourcePage` will fail. That's why the previous implementation was called `tc.ListApps` inside a `tc.RetryWithLogin`.

Clients created with `ConnectToCluster` will work as expected because their TLS configuration is always fetched from the `TeleportClient`.

So those functions that perform the app listing should still receive the `TeleportClient` + do the listing within `RetryWithLogin`. Alternatively, we could create the cluster client and perform a gRPC call to initialize the connection.

Reply:
That said, we could retrieve the profile (within login retry) and the cluster client outside it (inside a new function or not). The `getAppInfo` would take both as arguments, as well as the `TeleportClient`, so we can perform the listing with login retry.

Reply:
I've updated the `getAppInfo` to receive `TeleportClient` (which is used later for fetching the app with login retry) and auth client.